polliwog/polyline/_polyline_object.py

```diff
@@ -599,9 +599,13 @@ def nearest(
                 result.append(transform_result(indices_of_nearest_segments))
             if ret_distances:
                 result.append(
-                    transform_result(distance_to_closest_points_of_segments)[
-                        indices_of_nearest_segments
-                    ]
+                    transform_result(
+                        np.take_along_axis(
```

Comment:
I think it would be good to handle the `take_along_axis` calls the same for t_values and distances. Think it's fine to put it here in the `if` but could you move the one for t_values too?

Reply:
done